core/src/main/scala/io/odin/loggers/AsyncLogger.scala

```diff
@@ -73,25 +73,18 @@ object AsyncLogger {
       implicit F: Async[F]
   ): Resource[F, Logger[F]] = {
 
-    val createQueue = maxBufferSize match {
-      case Some(value) =>
-        Queue.bounded[F, LoggerMessage](value)
-      case None =>
-        Queue.unbounded[F, LoggerMessage]
-    }
-
     // Run internal loop of consuming events from the queue and push them down the chain
     def backgroundConsumer(logger: AsyncLogger[F]): Resource[F, Unit] = {
 
-      def drainLoop: F[Unit] = F.andWait(logger.drain, timeWindow).foreverM
+      def drainLoop: F[Unit] = F.delayBy(F.uncancelable(_ => logger.drain), timeWindow).foreverM
```

Comment:
It's not the primary concern of this PR, but might add useful context: I've seen very high resource usage caused by the timed approach to draining the buffer. Especially as https://github.com/scalafreaks/odin/blob/main/core/src/main/scala/io/odin/syntax/package.scala#L55 defines a very low delay of one millisecond, it ends up putting high pressure on the scheduler. At $work, that lead to idle pods using 80 milli-CPU – once I replaced the drain loop with a self-written fs2 version, they went down to 3 mCPU despite having database connections, HTTP server, etc.

So maybe this should be kept a `Queue` and instead of regularly trying to empty it, we could just semantically block when dequeuing (`take`) from an empty queue and be woken up when a new log message/operation is enqueued? If batching needs to be retained, one could follow the `take` with a `tryTakeN(None)` to check for more, then handle the result and `take` again.

Reply:
Can you open a separated issue to follow progress there? 

I like your suggestion, and I'd like to allow for more configurable behavior regarding the queue. In particular, I'd like to allow blocking for full queue instead of discarding, or discard old logs instead of new ones. 

I think these changes fit better in a separated PR and will be closely related with yours.